core/src/avm1/object/script_object.rs

```diff
@@ -429,18 +435,22 @@ impl<'gc> TObject<'gc> for ScriptObject<'gc> {
         }
     }
 
-    fn proto(&self) -> Value<'gc> {
-        self.0.read().prototype
+    fn proto(&self, activation: &mut Activation<'_, 'gc, '_>) -> Value<'gc> {
+        self.get_data("__proto__", activation)
     }
 
-    fn set_proto(&self, gc_context: MutationContext<'gc, '_>, prototype: Value<'gc>) {
-        self.0.write(gc_context).prototype = prototype;
+    fn set_proto(
+        &self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        prototype: Value<'gc>,
+    ) -> Result<(), Error<'gc>> {
+        self.set_data("__proto__", prototype, activation)
     }
```

Comment:
This method is only called in one place, in `Player`, so perhaps we should delete this method and move it inline. (`SuperObject` has a `set_proto`, but it should be unused/taken care of by nature of being a normal property.)

Reply:
Should the `set` in `Player` invoke virtual setters on the prototype chain? If not then we need to expose `set_data` through `TObject`, which is just an exchange of `set_proto` with `set_data`.

Reply:
I don't think this would consider prototype setters because the code in `Player` runs when a display object is initially built.

Reply:
If so we can use `define_value` there, which also cannot fail.